app.rb

```diff
@@ -2,18 +2,21 @@
 require 'pg'
 require_relative 'lib/peeps'
 require_relative 'db_connection_helper'
+require_relative 'lib/user'
+require_relative 'lib/db_connection'
 
 class Chitter < Sinatra::Base
 
   enable :sessions, :method_override
 
   get '/' do
+    @user = User.find_name_by_id(session[:user_id])
     erb(:index)
   end
 
   post '/login' do
-    session[:username] = params[:username]
-    session[:password] = params[:password]
+    user = User.authenticate(params[:email],params[:password])
+    session[:user_id] = user.first['id'] unless user.first == nil
```

Comment:
Why do you have to use .first here? Is there an other way you could have written the user file with which you are able to find the id in a more efficient way?